### PR TITLE
boards: shields: Correct active state of grant GPIO pin

### PR DESCRIPTION
**Pull request by rado17**
Correct active state of grant GPIO pin.

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]

**Comment by carlescufi** [comment]
@anangl please take a look

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/rpu_hw_if.c [hunk 1]
```diff
@@ -198,19 +208,26 @@ int rpu_pwron(void)
...
 	}
-	/* Settling time for iovdd nRF7002 DK/EK - switch (TCK106AG): ~600us
-	 * For nRF7002 Evaluation board, we add 1ms more to ensure
-	 * that the total time after bucken assertion is more than 2.5ms
-	 * so making this 3ms (1ms + 2ms)
+
+	/* Settling time for iovdd nRF7002 DK/EK - switch (TCK106AG): ~600us */
+	k_msleep(1);
+
+#ifdef SHIELD_NRF7002EB
```
Comment (anangl) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/rpu_hw_if.c:215]:
```suggestion
#ifdef CONFIG_SHIELD_NRF7002EB
```

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/rpu_hw_if.c [hunk 2]
```diff
@@ -35,6 +35,14 @@ GPIO_DT_SPEC_GET(NRF7002_NODE, iovdd_ctrl_gpios);
 static const struct gpio_dt_spec bucken_spec =
 GPIO_DT_SPEC_GET(NRF7002_NODE, bucken_gpios);
 
+#if defined(CONFIG_BOARD_NRF7000DK_NRF5340_CPUAPP) || \
+	defined(CONFIG_BOARD_NRF7001DK_NRF5340_CPUAPP) || \
+	defined(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP)
+#define NRF_RADIO_COEX_NODE DT_NODELABEL(nrf_radio_coex)
+static const struct gpio_dt_spec btrf_switch_spec =
+GPIO_DT_SPEC_GET(NRF_RADIO_COEX_NODE, btrf_switch_gpios);
+#endif /* CONFIG_BOARD_NRF700XDK_NRF5340_CPUAPP */
```
Comment (anangl) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/rpu_hw_if.c:44]: Options `BOARD_NRF7000DK_NRF5340_CPUAPP` and `BOARD_NRF7001DK_NRF5340_CPUAPP` do not exist, so this won't work as expected.
But wouldn't it be better to compile this conditionally when the `btrf_switch_gpios` property is present? I mean something like:
```
#define NRF_RADIO_COEX_NODE DT_NODELABEL(nrf_radio_coex)
#if DT_NODE_HAS_PROP(NRF_RADIO_COEX_NODE, btrf_switch_gpios)
static const struct gpio_dt_spec btrf_switch_spec =
	GPIO_DT_SPEC_GET(NRF_RADIO_COEX_NODE, btrf_switch_gpios);
#endif
```

**Review by anangl — COMMENTED** [review]
I think this PR needs a follow-up to address the following comments:

**Pull request merged**